docs/_tooling/extensions/score_metamodel/checks/check_options.py

```diff
@@ -38,80 +81,44 @@ def check_options(
     needs_types: list[NeedsInfoType] = None,
 ):
     """
-    Checking if all described and wanted attributes are present and their values follow the described pattern.
+    Checks that required and optional options and links are present and follow their defined patterns.
     """
     production_needs_types = app.config.needs_types
     if not needs_types:
         needs_types = production_needs_types
+
     try:
         need_options = get_need_type(needs_types, need["type"])
     except ValueError:
-        msg = "no type info defined for semantic check."
-        log.warning_for_option(need, "type", msg)
+        log.warning_for_option(need, "type", "no type info defined for semantic check.")
         return
 
-    required_options: dict[str, str] = need_options.get("mandatory_options", {})
-    optional_options: dict[str, str] = need_options.get("opt_opt", {})
-
-    if len(required_options) == 0:
-        msg = "no type info defined for semantic check."
-        log.warning_for_option(need, "type", msg)
-
-    for option, pattern in required_options.items():
-        values = need.get(option, None)
-
-        if values is None or values in [[], ""]:
-            msg = f"is missing required option: `{option}`."
-            log.warning_for_need(need, msg)
-            continue
-
-        # Normalize values (list or string)
-        if not isinstance(values, list):
-            values = [values]
-
-        for value in values:
-            assert isinstance(value, str)
-            regex = re.compile(pattern)
-
-            if not regex.match(value):
-                msg = f"does not follow pattern `{pattern}`."
-                log.warning_for_option(need, option, msg)
+    if not need_options.get("mandatory_options", {}):
+        log.warning_for_option(need, "type", "no type info defined for semantic check.")
 
-    for option, pattern in optional_options.items():
-        values = need.get(option, None)
-
-        if values is None or values in [[], ""]:  # Skip if empty
-            continue
-
-        if not isinstance(values, list):
-            values = [values]
+    # Validate Options
+    validate_fields(
+        need,
+        log,
+        need_options.get("mandatory_options", {}),
+        required=True,
+        context="option",
+    )
```

Comment:
Just a suggestion how to condense this even further. Although it probably suffers from less readability in this style.

```suggestion
    # Validate Options
    checking_dict = {"option": ["mandatory_options", "opt_opt"], "link": ["mandatory_links", "opt_link"]}
    for context, check_fields in checking_dict.items():
    for field in check_fiels:
       validate_fields(
          need,
          log,
          need_options.get(check_field, {}),
          required=True,
          context=context,
       )
```

Reply:
Fixed. Please resolve.